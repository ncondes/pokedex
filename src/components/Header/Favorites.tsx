'use client'

import Link from 'next/link'
import { FC } from 'react'
import { usePathname } from 'next/navigation'

export const Favorites: FC = () => {
   const pathname = usePathname()

   return (
      <Link aria-label="favorites link" href="/favorites">
         {pathname === '/favorites' ? (
            <i className="fa-solid fa-heart fa-lg"></i>
         ) : (
            <i className="fa-regular fa-heart fa-lg"></i>
         )}
      </Link>
   )
}
