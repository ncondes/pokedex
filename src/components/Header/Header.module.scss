@use '@/scss/colors' as *;

.header {
   height: 3.5rem;
   background-color: $blue-primary-color;
   padding: 0 1.5rem;

   ul {
      display: flex;
      justify-content: space-between;
      align-items: center;
      height: 100%;
      list-style: none;
   }
}

.home_link {
   display: flex;
   align-items: center;
   height: 100%;

   span {
      margin-left: 0.5rem;
      color: $yellow-secondary-color;
      font-size: 1.5rem;
      font-weight: 500;
      text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
   }
}

.theme_and_favorites_links {
   display: flex;
   justify-content: center;
   align-items: center;

   li {
      margin-left: 1rem;
   }

   button {
      background-color: transparent;
   }

   i {
      color: $white-cream-color;
   }
}
