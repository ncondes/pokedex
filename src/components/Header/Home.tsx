import Image from 'next/image'
import Link from 'next/link'
import { FC } from 'react'
import styles from './Header.module.scss'

export const Home: FC = () => {
   return (
      <Link href="/" className={styles.home_link}>
         <div>
            <Image
               src={'/images/pokeball-icon-color.png'}
               alt="pokeball icon"
               width={25}
               height={25}
            />
         </div>
         <span>Pokedex</span>
      </Link>
   )
}
