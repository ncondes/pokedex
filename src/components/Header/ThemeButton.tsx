'use client'

import { ThemeContext } from '@/context/theme'
import { FC, useContext } from 'react'

export const ThemeButton: FC = () => {
   const { theme, toggleTheme } = useContext(ThemeContext)

   const handleClick = () => toggleTheme()

   return (
      <button onClick={handleClick}>
         <i
            className={`fa-regular ${
               theme === 'light' ? 'fa-moon' : 'fa-sun'
            } fa-xl`}
         ></i>
      </button>
   )
}
