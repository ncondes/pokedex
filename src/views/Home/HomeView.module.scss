@use '@/scss/mixins' as *;

.tools {
   margin: 2rem auto;
   display: grid;
   grid-template-rows: repeat(3, 1fr);
   grid-template-columns: 1fr;

   @include breakpoint-up('md') {
      grid-template-rows: repeat(2, 1fr);
      grid-template-columns: repeat(2, 1fr);
      grid-gap: 0 1rem;
   }

   @include breakpoint-up('lg') {
      grid-template-rows: 1fr;
      grid-template-columns: 4fr 1fr 1fr;
      grid-gap: 1rem;
   }
}

.introduction {
   margin-bottom: 1rem;
   text-align: center;
   font-size: 1rem;
   font-weight: 300;
}

.introduction_extended {
   display: none;

   @include breakpoint-up('md') {
      display: block;
   }
}
