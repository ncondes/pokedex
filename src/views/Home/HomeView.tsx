'use client'

import { SearchBox } from '@/components/SearchBox'
import { Select } from '@/components/Select'
import { Pokemon } from '@/interfaces'
import { FC, useContext } from 'react'
import { useFilteredPokemons } from '@/hooks'
import { PokemonList } from '@/components/PokemonList'
import { ThemeContext } from '@/context/theme'
import styles from './HomeView.module.scss'

interface Props {
   data: Pokemon[]
}

const pokemonTypes = [
   'all',
   'bug',
   'dark',
   'dragon',
   'electric',
   'fairy',
   'fighting',
   'fire',
   'flying',
   'ghost',
   'grass',
   'ground',
   'ice',
   'normal',
   'poison',
   'psychic',
   'rock',
   'steel',
   'water',
]

const sortOptions = ['ascending', 'descending']

export const HomeView: FC<Props> = ({ data }) => {
   const { pokemons, handleSearch, handleFilter, handleSort } =
      useFilteredPokemons(data)

   const { useTheme } = useContext(ThemeContext)

   return (
      <>
         <p className={`${styles.introduction} ${useTheme('font')}`}>
            Discover and explore a vast collection of diverse and fascinating
            Pokemon creatures.{' '}
            <span className={styles.introduction_extended}>
               Easily search for your favorite Pokemon, learn about their unique
               abilities, and uncover their captivating stories. Start your
               adventure now and delve into the magical world of Pokemon!
            </span>
         </p>
         <div className={styles.tools}>
            <SearchBox callback={handleSearch} />
            <Select
               label="filter"
               icon="fa-filter"
               options={pokemonTypes}
               onChange={handleFilter}
            />
            <Select
               label="sort"
               icon="fa-sort"
               options={sortOptions}
               onChange={handleSort}
            />
         </div>
         <PokemonList pokemons={pokemons} />
      </>
   )
}
