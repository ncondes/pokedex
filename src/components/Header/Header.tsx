import { ThemeButton } from './ThemeButton'
import { Favorites } from './Favorites'
import styles from './Header.module.scss'
import { Home } from './Home'

export const Header = () => {
   return (
      <header className={styles.header}>
         <ul>
            <li>
               <Home />
            </li>
            <div className={styles.theme_and_favorites_links}>
               <li>
                  <ThemeButton />
               </li>
               <li>
                  <Favorites />
               </li>
            </div>
         </ul>
      </header>
   )
}
