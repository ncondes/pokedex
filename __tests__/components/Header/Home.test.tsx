import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Home } from '@/components/Header/Home'

describe('Home', () => {
   it('renders the correct elements with the proper props', () => {
      const { getByText, getByAltText, container } = render(<Home />)
      const link = container.querySelector('a')

      expect(getByText('Pokedex')).toBeInTheDocument()
      expect(getByAltText('pokeball icon')).toBeInTheDocument()
      expect(link).toHaveAttribute('href', '/')
   })

   it('should match the snapshot', () => {
      const { container } = render(<Home />)

      expect(container).toMatchSnapshot()
   })
})
