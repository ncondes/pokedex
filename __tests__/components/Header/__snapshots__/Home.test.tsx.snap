// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<div>
  <a
    class="home_link"
    href="/"
  >
    <div>
      <img
        alt="pokeball icon"
        data-nimg="1"
        decoding="async"
        height="25"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fpokeball-icon-color.png&w=64&q=75"
        srcset="/_next/image?url=%2Fimages%2Fpokeball-icon-color.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Fpokeball-icon-color.png&w=64&q=75 2x"
        style="color: transparent;"
        width="25"
      />
    </div>
    <span>
      Pokedex
    </span>
  </a>
</div>
`;
